test/devicecontroller/DefaultDeviceController.test.ts

```diff
@@ -2408,48 +2415,182 @@ describe('DefaultDeviceController', () => {
+
+      expect(spy1.called).to.be.false;
+      expect(spy2.called).to.be.false;
+      expect(spy3.called).to.be.false;
+      expect(gUM.called).to.be.false;
+    });
+
+    it('connects the transform video stream to the video element if there is a transform device stream', async () => {
+      // It is important that the domMockBehavior setup is before the original video stream generation to mock browser behavior.
+      domMockBehavior.createElementCaptureStream = new MediaStream();
+      domMockBehavior.mediaStreamTrackSettings.deviceId = stringDeviceId;
+      domMockBehavior.createElementCaptureStream.addTrack(new MediaStreamTrack());
+      // Connect the original video stream
       await deviceController.chooseVideoInputDevice(stringDeviceId);
       deviceController.startVideoPreviewForVideoInput(element);
+      const processor = new NoOpVideoFrameProcessor();
+      const device = new DefaultVideoTransformDevice(logger, stringDeviceId, [processor]);
+      // Connect the transform video stream
+      await deviceController.chooseVideoInputDevice(device);
+      deviceController.startVideoPreviewForVideoInput(element);
+
+      expect(device.getInnerDevice() === stringDeviceId);
+      expect(element.srcObject === (await gUM.returnValues.pop()));
+      expect(spy1.called).to.be.false;
+      expect(spy2.called).to.be.false;
+      expect(spy3.calledTwice).to.be.true;
+      expect(gUM.calledOnce).to.be.true;
+      await device.stop();
+    });
+
+    it('connects the second transform video stream if a transform stream exists', async () => {
+      class TestVideoFrameProcessor1 extends NoOpVideoFrameProcessor {
+        width = 0;
+        height = 0;
+        process(_buffers: VideoFrameBuffer[]): Promise<VideoFrameBuffer[]> {
+          this.width = 10;
+          this.height = 10;
+          return Promise.resolve(_buffers);
+        }
+      }
+
+      class TestVideoFrameProcessor2 extends NoOpVideoFrameProcessor {
+        width = 0;
+        height = 0;
+        process(_buffers: VideoFrameBuffer[]): Promise<VideoFrameBuffer[]> {
+          this.width = 20;
+          this.height = 20;
+          return Promise.resolve(_buffers);
+        }
+      }
+      domMockBehavior.createElementCaptureStream = new MediaStream();
+      domMockBehavior.mediaStreamTrackSettings.deviceId = stringDeviceId;
+      domMockBehavior.createElementCaptureStream.addTrack(new MediaStreamTrack());
+      audioVideoController = new NoOpAudioVideoController();
+      const processor = new TestVideoFrameProcessor1();
+      const firstTransformDevice = new DefaultVideoTransformDevice(logger, stringDeviceId, [
+        processor,
+      ]);
+      await deviceController.chooseVideoInputDevice(firstTransformDevice);
+      deviceController.startVideoPreviewForVideoInput(element);
+      const firstMediaStream = await gUM.returnValues[0];
+      // In the unit test, there is no UI so no call to openVideoInputFromSelection is made.
```

Comment:
This is not related to UI. In browser, calling stop stream track will make the stream to inactive. our dom mock builder does not implement this behavior.

Reply:
How about:
> The current DOMMockBuilder does not set active property of media stream to false if the media stream track is stopped which is not consistent to the browser behavior so we have to manually set the property here.